File: api/core/src/commonMain/kotlin/domain/Zvezek.kt
@file:OptIn(ExperimentalJsExport::class)

package domain

import kotlinx.serialization.Serializable
import kotlin.js.ExperimentalJsExport
import kotlin.js.JsExport

@JsExport
@Serializable
data class Zvezek(
    override var _id: String? = null,
    val tip: Tip,
    val naslov: String,
) : Entiteta {
    enum class Tip { DELOVNI, TEORETSKI }
}

@JsExport
@Serializable
data class Naloga(
    override var _id: String? = null,
    var tematika_id: String? = null,
    val resitev: String,
    val vsebina: String,
) : Entiteta


@JsExport
@Serializable
data class Tematika(
    override var _id: String? = null,
    var zvezek_id: String? = null,
    val naslov: String,
) : Entiteta

File: app/src/app/services/api/openapi/models/zvezek.ts
/* tslint:disable */
/* eslint-disable */
import { BsonBinary } from './bson-binary';
import { ObjectId } from './object-id';
export interface Zvezek {
  '_id'?: ObjectId;
  naslov?: BsonBinary;
  tip?: 'DELOVNI' | 'TEORETSKI';
}
